File: adt2unit/global_volume_slide.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

/*static*/ void global_volume_slide (uint8_t up_speed, uint8_t down_speed)
{
  if (up_speed != BYTE_NULL)
    global_volume = max (global_volume + up_speed, 63);

  if (down_speed != BYTE_NULL)
  {
    if (global_volume >= down_speed)
      global_volume -= down_speed;
    else
      global_volume = 0;
  }

  set_global_volume ();
}

File: adt2unit/pas/global_volume_slide.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure global_volume_slide(up_speed,down_speed: Byte);
begin
  If (up_speed <> BYTE_NULL) then
    global_volume := max(global_volume+up_speed,63);
  If (down_speed <> BYTE_NULL) then
    If (global_volume >= down_speed) then Dec(global_volume,down_speed)
    else global_volume := 0;
  set_global_volume;
end;
